polar-core/src/error.rs

```diff
@@ -298,6 +298,7 @@ pub enum RuntimeError {
         var: Symbol,
         term: Term,
     },
+    DataFilteringError(String),
```

Comment:
We're trying to get a little better with how we construct errors. Instead of constructing the error message at the callsite, you should store the relevant info in the error itself, and construct the message as part of `impl fmt::Display for RuntimeError`

So this would be:

```
DataFilteringError { var_type: String, field: String }
```
and the message formatting would be handled on 361: https://github.com/osohq/oso/pull/1330/files#diff-20681b7129ace19298fd33795d95bbf7c5b118c4be2bf33844520b21d8798d45R361